app/views/projects/_find_file_link.html.haml

```diff
@@ -0,0 +1,6 @@
+- if project && project_namespace && ref

```

Comment:
We can just use `@project` and `@ref` directly here, and we don't need to verify that they exist since they always do on the tree page :)